pkg/provider/doppler/doppler.go

```diff
@@ -0,0 +1,305 @@
+		if err := client.SetBaseURL(customBaseURL); err != nil {
+			return nil, fmt.Errorf(errNewClient, err)
+		}
+	}
+
+	if customVerifyTLS, found := os.LookupEnv(verifyTLSOverrideEnvVar); found {
+		customVerifyTLS, err := strconv.ParseBool(customVerifyTLS)
+		if err == nil {
+			client.VerifyTLS = customVerifyTLS
+		}
+	}
+
+	d.client = client
+	d.project = kubeClient.store.Project
+	d.config = kubeClient.store.Config
+	d.nameTransformer = kubeClient.store.NameTransformer
+	d.format = kubeClient.store.Format
+
+	return d, nil
+}
+
+func (d *Doppler) Validate() (esv1beta1.ValidationResult, error) {
+	timeout := 15 * time.Second
+	clientURL := d.client.BaseURL().String()
+
+	if err := utils.NetworkValidate(clientURL, timeout); err != nil {
+		return esv1beta1.ValidationResultError, err
+	}
+
+	if err := d.client.Authenticate(); err != nil {
+		return esv1beta1.ValidationResultError, err
+	}
+
+	return esv1beta1.ValidationResultReady, nil
+}
+
+func (d *Doppler) ValidateStore(store esv1beta1.GenericStore) error {
+	storeSpec := store.GetSpec()
+	dopplerStoreSpec := storeSpec.Provider.Doppler
+	dopplerTokenSecretRef := dopplerStoreSpec.Auth.SecretRef.DopplerToken
+	if err := utils.ValidateSecretSelector(store, dopplerTokenSecretRef); err != nil {
+		return fmt.Errorf(errInvalidStore, err)
+	}
+
+	if dopplerTokenSecretRef.Name == "" {
+		return fmt.Errorf(errInvalidStore, "dopplerToken.name cannot be empty")
+	}
+
+	return nil
+}
+
+func (d *Doppler) GetSecrets(_ context.Context) (map[string][]byte, error) {
+	request := client.SecretsRequest{
+		Project:         d.project,
+		Config:          d.config,
+		NameTransformer: d.nameTransformer,
+		Format:          d.format,
+	}
+
+	response, err := d.client.GetSecrets(request)
+	if err != nil {
+		return nil, fmt.Errorf(errGetSecrets, err)
+	}
+
+	if d.format != "" {
+		return map[string][]byte{
+			secretsDownloadFileKey: response.Body,
+		}, nil
+	}
+
+	return ExternalSecretsFormat(response.Secrets), nil
+}
+
+func (d *Doppler) GetSecret(_ context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	request := client.SecretRequest{
+		Name:    ref.Key,
+		Project: d.project,
+		Config:  d.config,
+	}
+
+	secret, err := d.client.GetSecret(request)
+	if err != nil {
+		return nil, fmt.Errorf(errGetSecret, ref.Key, err)
+	}
+
+	return []byte(secret.Value), nil
+}
+
+func (d *Doppler) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	data, err := d.GetSecret(ctx, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	kv := make(map[string]json.RawMessage)
+	err = json.Unmarshal(data, &kv)
+	if err != nil {
+		return nil, fmt.Errorf(errUnmarshalSecretMap, ref.Key, err)
+	}
+
+	secretData := make(map[string][]byte)
+	for k, v := range kv {
+		var strVal string
+		err = json.Unmarshal(v, &strVal)
+		if err == nil {
+			secretData[k] = []byte(strVal)
+		} else {
+			secretData[k] = v
+		}
```

Comment:
I'm not clear on why the unmarshalling to a string is happening but this code is in practically every provider so figured there must be a good reason.